src/main/java/codeu/model/data/Message.java

```diff
@@ -63,6 +71,48 @@ public String getContent() {
     return content;
   }
 
+  /** Finds picture links that end in jpg and add to the end of the message the
+   * html rendered picture (link picture, opens to picture in another window)
+   * with maximum picture width at 500 */
+  public String parseImages(String message) { 
+    int startIndex = 0;
+    int linkStartIndex;
+    int spaceIndex;
+    int newlineIndex;
+    int endWordIndex;
+    List<String> links = new ArrayList<String>(); 
+
+    while (startIndex < message.length()) {
+        linkStartIndex = message.indexOf("http", startIndex);
+        if(linkStartIndex == -1)
+            return message;   //indicated http not in string
+ 
+        spaceIndex = message.indexOf(' ', linkStartIndex);
+        newlineIndex = message.indexOf('\n', linkStartIndex);
+        if(spaceIndex == -1)     //-1 if no space in the rest of the string
+            spaceIndex = message.length();
+        if(newlineIndex == -1)
+            newlineIndex = message.length();
+        endWordIndex = Math.min(newlineIndex, spaceIndex);
+        System.out.println("start index is " + startIndex + " end is " + endWordIndex);
+
+        if(message.substring(endWordIndex - 4, endWordIndex).equals(".jpg")) {
+            links.add("\n<a href =" + message.substring(
```

Comment:
Instead of adding the links to the message.  Can you instead return a list of image URLs here?  We should abstract the Images concept in our classes.  The conversion from a URL to a <img> link shouldn't happen until rendering is needed (in the JSP.)

Reply:
But then wouldn't we be doing this entire string parsing thing a lot? I don't think the string parsing method would be very efficient with long messages